django/db/models/enums.py

```diff
@@ -56,37 +76,47 @@ def values(cls):
         return [value for value, _ in cls.choices]
 
 
-class Choices(enum.Enum, metaclass=ChoicesMeta):
+class Choices(enum.Enum, metaclass=ChoicesType):
     """Class for creating enumerated choices."""
 
-    @DynamicClassAttribute
-    def label(self):
-        return self._label_
+    if PY311:
+        do_not_call_in_templates = enum.nonmember(True)
+    else:
 
-    @property
-    def do_not_call_in_templates(self):
-        return True
+        @property
+        def do_not_call_in_templates(self):
+            return True
 
-    def __str__(self):
-        """
-        Use value when cast to str, so that Choices set as model instance
-        attributes are rendered as expected in templates and similar contexts.
-        """
-        return str(self.value)
+    @enum_property
+    def label(self):
+        return self._label_
 
     # A similar format was proposed for Python 3.10.
     def __repr__(self):
         return f"{self.__class__.__qualname__}.{self._name_}"
 
 
-class IntegerChoices(int, Choices):
+class IntegerChoices(Choices, IntEnum):
     """Class for creating enumerated integer choices."""
 
-    pass
-
 
-class TextChoices(str, Choices):
+class TextChoices(Choices, StrEnum):
     """Class for creating enumerated string choices."""
 
+    @staticmethod
     def _generate_next_value_(name, start, count, last_values):
         return name
+
+
+def __getattr__(name):
+    if name == "ChoicesMeta":
+        import warnings
+
+        from django.utils.deprecation import RemovedInDjango60Warning
+
+        warnings.warn(
+            "ChoicesMeta has been renamed to ChoicesType.",
+            RemovedInDjango60Warning,
+        )
```

Comment:
```suggestion
        warnings.warn(
            "ChoicesMeta has been renamed to ChoicesType.",
            RemovedInDjango60Warning,
            stacklevel=2,
        )
```

See B028 on https://github.com/PyCQA/flake8-bugbear#list-of-warnings